effekt/shared/src/main/scala/effekt/Compiler.scala

```diff
@@ -125,6 +125,15 @@ trait Compiler[Executable] {
       mod
     }
 
+  /**
+   * Used by the server to typecheck, report type errors and show
+   * captures at boxes and definitions 
+   */
+  def runMiddleend(source: Source)(using Context): Option[Module] =
+    val typechecked = Frontend(source)
+    typechecked.foreach(res => validate(source, res.mod))
+    typechecked.flatMap(Middleend.run).map(_.mod)
```

Comment:
Sorry for nitpicking so much, but isn't this roughly equivalent to:
```
(Frontend andThen Middleend).map { res => validate(res.source, res.mod); res.mod }
 ```

Reply:
No worries. If it doesn't matter when the `validate` function is called, then this should equivalent, true. I thought it might be important to validate right after the frontend.

Reply:
Maybe I am missing something, but the most important thing is that validate is called as part of the LSP service